shared/js/media/media_frame.js

```diff
@@ -51,6 +51,7 @@ function MediaFrame(container, includeVideo, maxImageSize) {
     this.video = new VideoPlayer(container);
     this.video.hide();
   }
+  navigator.mozL10n.once(this.localize.bind(this));
```

Comment:
I think that this is what I suggested you do, but I'm having second thoughts.  If the l10n data is ready, this may call localieImage before displayImage() has been called, there may not be anything to be localized.  And if the l10n data is not ready yet, then this plus the ready() call in frames.js may mean that localize gets called twice.

Given the call to localize() in  displayImage(), I think that this once() call isn't actually necessary.